packages/core/src/use-cases/inventory-movements/report-annual-outbound-inventorymovements-use-case.ts

```diff
@@ -0,0 +1,78 @@
+import type { InventoryMovement } from '@stocker/core/entities'
+import type { InventoryMovementDto } from '../../dtos'
+import { NotAllowedError, NotFoundError } from '../../errors'
+import type {
+  IBatchesRepository,
+  IInventoryMovementsRepository,
+  IProductsRepository,
+} from '../../interfaces'
+import dayjs from 'dayjs'
+
+type Request = {
+  productId?: string
+}
+
+export class ReportAnnualOutboundInventorymovementsUseCase {
+  private readonly inventoryMovementsRepository: IInventoryMovementsRepository
+  private readonly productsRepository: IProductsRepository
+
+  constructor(
+    inventoryMovementsRepository: IInventoryMovementsRepository,
+    productsRepository: IProductsRepository,
+  ) {
+    this.inventoryMovementsRepository = inventoryMovementsRepository
+    this.productsRepository = productsRepository
+  }
+
+  async execute({ productId }: Request) {
+    try {
+      if (productId) {
+        const product = await this.productsRepository.findById(productId)
+        if (!product) {
+          throw new NotFoundError('Product not found')
+        }
+      }
+
+      const inventoryMovements: InventoryMovement[] = await this.inventoryMovementsRepository.findByDateRange({
+        productId,
+      });
+
+      const formattedMovements = this.formatByMonth(inventoryMovements);
+      
+      return { months: formattedMovements }
+      
+    } catch (error) {
+      console.error('Error in use case execution:', error)
+      throw error
+    }
+  }
+
+  private formatByMonth(inventoryMovements: InventoryMovement[]) {
+    const months = [
```

Comment:
Months should be generated  dinamically and not hard coded, so if we are in October the month s would start with Novemeber, December, January, February etc...